osu.Game/Overlays/Toolbar/ToolbarClock.cs

```diff
@@ -0,0 +1,323 @@
+
+        private const float hand_thickness = 2.4f;
+
+        public ToolbarClock()
+        {
+            RelativeSizeAxes = Axes.Y;
+            AutoSizeAxes = Axes.X;
+
+            Padding = new MarginPadding(5);
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            clockDisplayMode = config.GetBindable<ToolbarClockDisplayMode>(OsuSetting.ToolbarClockDisplayMode);
+
+            InternalChild = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.Y,
+                AutoSizeAxes = Axes.X,
+                Direction = FillDirection.Horizontal,
+                Spacing = new Vector2(5),
+                Children = new Drawable[]
+                {
+                    analog = new AnalogDisplay
+                    {
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                    },
+                    digital = new DigitalDisplay
+                    {
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            clockDisplayMode.BindValueChanged(displayMode =>
+            {
+                bool showAnalog = displayMode.NewValue == ToolbarClockDisplayMode.Analog || displayMode.NewValue == ToolbarClockDisplayMode.Full;
+                bool showDigital = displayMode.NewValue != ToolbarClockDisplayMode.Analog;
+                bool showRuntime = displayMode.NewValue == ToolbarClockDisplayMode.DigitalWithRuntime || displayMode.NewValue == ToolbarClockDisplayMode.Full;
+
+                digital.FadeTo(showDigital ? 1 : 0);
+                digital.ShowRuntime = showRuntime;
+
+                analog.FadeTo(showAnalog ? 1 : 0);
+            }, true);
+        }
+
+        protected override bool OnClick(ClickEvent e)
+        {
+            cycleDisplayMode();
+            return true;
+        }
+
+        private void cycleDisplayMode()
+        {
+            switch (clockDisplayMode.Value)
+            {
+                case ToolbarClockDisplayMode.Analog:
+                    clockDisplayMode.Value = ToolbarClockDisplayMode.Full;
+                    break;
+
+                case ToolbarClockDisplayMode.Digital:
+                    clockDisplayMode.Value = ToolbarClockDisplayMode.Analog;
+                    break;
+
+                case ToolbarClockDisplayMode.DigitalWithRuntime:
+                    clockDisplayMode.Value = ToolbarClockDisplayMode.Digital;
+                    break;
+
+                case ToolbarClockDisplayMode.Full:
+                    clockDisplayMode.Value = ToolbarClockDisplayMode.DigitalWithRuntime;
+                    break;
+            }
+        }
+
+        private class DigitalDisplay : ClockDisplay
```

Comment:
nit: I personally find it better to have `DigitalDisplay` and `AnalogDisplay` both stored in their own files as partials of `ToolbarClock` (i.e. `ToolbarClock_DigitalDisplay.cs` and `ToolbarClock_AnalogDisplay.cs`), as the number of private classes stored in here are somewhat obscuring the logic behind how this is working.

Reply:
I've moved them to their own files